packages/x-data-grid/src/models/props/DataGridProps.ts

```diff
@@ -459,7 +459,7 @@ export interface DataGridPropsWithoutDefaultValue<R extends GridValidRowModel =
    * @param {GridCellParams} params With all properties from [[GridCellParams]].
    * @returns {boolean} A boolean indicating if the cell is editable.
    */
-  isCellEditable?: (params: GridCellParams<any, R>) => boolean;
+  isCellEditable?: (params: GridCellParams<R>) => boolean;
```

Comment:
It indeed seems an incorrect way to use the type since the users providing the `R` value won't be able to get the proper type checking.

I think `getCellClassName` could also benefit from a similar [change](https://github.com/mui/mui-x/pull/12896/commits/0d7f1566dabe19b29dfb372e7dce16a7b5759716).

It could possibly start showing errors to some users, but IMO, it's a bugfix rather than a BC, since the user's expectation would be to actually get a TS error in case of unwanted key access.

Any disagreements @mui/xgrid?